packages/contracts/test/framework/plugin/plugin-setup-processor.ts

```diff
@@ -878,340 +1238,676 @@ describe.skip('Plugin Setup Processor', function () {
+        preparedSetupData: {
+          permissions: secondPreparationPermissions,
+          helpers: secondPreparationHelpers,
+        },
+        initData: secondPreparationInitData,
+        preparedSetupId,
+      } = await prepareUpdate(
+        psp,
+        targetDao.address,
+        proxy,
+        currentVersion,
+        secondPreparationNewVersion,
+        currentPluginRepoPointer[0],
+        helpersUV1,
+        EMPTY_DATA
+      );
 
-        await expect(
-          psp.applyUpdate(
-            targetDao.address,
+      await expect(
+        psp.callStatic.applyUpdate(
+          targetDao.address,
+          createApplyUpdateParams(
             proxy,
-            setupUV2.address,
-            repoU.address,
-            initDataV2,
-            permissionsV2
+            newPluginRepoPointer,
+            firstPreparationInitData,
+            firstPreparationPermissions,
+            firstPreparationHelpers
           )
         )
-          .to.be.revertedWithCustomError(targetDao, 'Unauthorized')
-          .withArgs(
-            targetDao.address,
-            permissionsV2[0]['where'],
-            psp.address,
-            ROOT_PERMISSION_ID
-          );
-      });
+      ).not.to.be.reverted;
 
-      it('reverts if the plugin setup processor does not have the `UPGRADE_PLUGIN_PERMISSION_ID` permission', async () => {
-        const pluginUpdateParams = {
-          plugin: proxy,
-          pluginSetupRepo: repoU.address,
-          currentPluginSetup: setupUV1.address,
-          newPluginSetup: setupUV2.address,
-        };
-        const prepareUpdateTx = await psp.prepareUpdate(
+      await expect(
+        psp.callStatic.applyUpdate(
           targetDao.address,
-          pluginUpdateParams,
-          helpersUV1,
-          EMPTY_DATA
-        );
-
-        const prepareUpdateEvent = await findEvent(
-          prepareUpdateTx,
-          EVENTS.UpdatePrepared
-        );
+          createApplyUpdateParams(
+            proxy,
+            [
+              newPluginRepoPointer[0],
+              secondPreparationNewVersion[0],
+              secondPreparationNewVersion[1],
+            ],
+            secondPreparationInitData,
+            secondPreparationPermissions,
+            secondPreparationHelpers
+          )
+        )
+      ).not.to.be.reverted;
 
-        const {permissions: permissionsV2, initData: initDataV2} =
-          prepareUpdateEvent.args;
+      // // Apply one of the preparation
+      await applyUpdate(
+        psp,
+        targetDao.address,
+        proxy,
+        [
+          newPluginRepoPointer[0],
+          secondPreparationNewVersion[0],
+          secondPreparationNewVersion[1],
+        ],
+        secondPreparationInitData,
+        secondPreparationPermissions,
+        secondPreparationHelpers
+      );
 
-        await expect(
-          psp.applyUpdate(
-            targetDao.address,
+      // confirm that now preparations can't be applied anymore
```

Comment:
check at least one of them as non-static so we can make sure we land in the rigth code path

Reply:
Removed `callStatic` check ! it was not needed.